log4j-nosql/src/main/java/org/apache/logging/log4j/nosql/appender/mongodb/MongoDbProvider.java

```diff
@@ -93,130 +103,249 @@ public String toString() {
+				.setFactoryMethodName(factoryMethodName).build();
+	}
+
+	@PluginBuilderFactory
+	public static <B extends Builder<B>> B newBuilder() {
+		return new Builder<B>().asBuilder();
+	}
+
+	public static class Builder<B extends Builder<B>> extends AbstractFilterable.Builder<B>
+			implements org.apache.logging.log4j.core.util.Builder<MongoDbProvider> {
+
+		@PluginBuilderAttribute
+		@ValidHost
+		private String server = "localhost";
+
+		@PluginBuilderAttribute
+		@ValidPort
+		private String port = "" + DEFAULT_PORT;
+
+		@PluginBuilderAttribute
+		@Required(message = "No database name provided")
+		private String databaseName;
+
+		@PluginBuilderAttribute
+		@Required(message = "No collection name provided")
+		private String collectionName;
+
+		@PluginBuilderAttribute
+		private String userName;
+
+		@PluginBuilderAttribute(sensitive = true)
+		private String password;
+
+		@PluginBuilderAttribute("capped")
+		private boolean isCapped = false;
+
+		@PluginBuilderAttribute
+		private int collectionSize = DEFAULT_COLLECTION_SIZE;
+
+		@PluginBuilderAttribute
+		private String factoryClassName;
+
+		@PluginBuilderAttribute
+		private String factoryMethodName;
+
+		@PluginBuilderAttribute
+		private String writeConcernConstantClassName;
+
+		@PluginBuilderAttribute
+		private String writeConcernConstant;
+
+		public B setServer(String server) {
+			this.server = server;
+			return asBuilder();
+		}
+
+		public B setPort(String port) {
+			this.port = port;
+			return asBuilder();
+		}
+
+		public B setDatabaseName(String databaseName) {
+			this.databaseName = databaseName;
+			return asBuilder();
+		}
+
+		public B setCollectionName(String collectionName) {
+			this.collectionName = collectionName;
+			return asBuilder();
+		}
+
+		public B setUserName(String userName) {
+			this.userName = userName;
+			return asBuilder();
+		}
+
+		public B setPassword(String password) {
+			this.password = password;
+			return asBuilder();
+		}
+
+		public B setCapped(boolean isCapped) {
+			this.isCapped = isCapped;
+			return asBuilder();
+		}
+
+		public B setCollectionSize(int collectionSize) {
+			this.collectionSize = collectionSize;
+			return asBuilder();
+		}
+
+		public B setFactoryClassName(String factoryClassName) {
+			this.factoryClassName = factoryClassName;
+			return asBuilder();
+		}
+
+		public B setFactoryMethodName(String factoryMethodName) {
+			this.factoryMethodName = factoryMethodName;
+			return asBuilder();
+		}
+
+		public B setWriteConcernConstantClassName(String writeConcernConstantClassName) {
+			this.writeConcernConstantClassName = writeConcernConstantClassName;
+			return asBuilder();
+		}
+
+		public B setWriteConcernConstant(String writeConcernConstant) {
+			this.writeConcernConstant = writeConcernConstant;
+			return asBuilder();
+		}
+        
+		@Override
+		public MongoDbProvider build() {
+	        DB database;
+	        String description;
```

Comment:
It's not a huge deal performance-wise, but you could use a StringBuilder here for the more advanced concatenation. I doubt javac can optimize it like it does with other string concatenations.